### PR TITLE
Name Lambda Params with Spans

### PR DESCRIPTION
**Pull request by ScottCarda-MS**
Lifted lambda parameter names should be named according to the parameters specified in the original lambda expression, along with having the appropriate spans. Otherwise this information is lost and unrecoverable.

### CODE REVIEW

**Review by kuzminrobin — APPROVED** [review]
Approving, but my approval will hardly unblock this PR since I'm not the code owner.

**Review by swernli — APPROVED** [review]
LGTM

**Pull request merged**
